flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -456,6 +456,12 @@ protected StreamTask(
 
             CheckpointStorageAccess checkpointStorageAccess =
                     checkpointStorage.createCheckpointStorage(getEnvironment().getJobID());
+            environment
+                    .getTaskStateManager()
+                    .initCheckpointFileSystem(
+                            checkpointStorageAccess,
+                            environment.getJobVertexId(),
+                            environment.getTaskInfo());
```

Comment:
A little bit strange here. I think it is better to provide a function to apply file-merging configuration and get a new `CheckpointStorageAccess` from the `environment.getTaskStateManager().getFileMergingSnapshotManager(xxx)` and `checkpointStorageAccess` here. The new `CheckpointStorageAccess` will access the checkpoint storage in file-merging way, leveraging the `FileMergingSnapshotManager`. However in this PR, the new `CheckpointStorageAccess` and corresponding `CheckpointStorageLocation` could just wrap existing classes and instance of `FileMergingSnapshotManager`, and only do proxy.

WDYT?

Reply:
👍 I created `FsMergingCheckpointStorageAccess` here, and `FileMergingSnapshotManager#initFileSystem` is called when `FsMergingCheckpointStorageAccess` is constructed.